basculehttp/bearerTokenFactory_test.go

```diff
@@ -234,3 +234,73 @@ good:
 		})
 	}
 }
+
+func TestProvideResolver(t *testing.T) {
+	type In struct {
+		fx.In
+		R clortho.Resolver `name:"key_resolver"`
+	}
+
+	const yaml = `
+good:
+  factory:
+    uri: "http://test:1111/keys/{keyId}"
+  purpose: 0
+  updateInterval: 604800000000000
+`
+	v := viper.New()
+	v.SetConfigType("yaml")
+	require.NoError(t, v.ReadConfig(strings.NewReader(yaml)))
+
+	goodResolver, err := clortho.NewResolver(clortho.WithKeyIDTemplate("good"))
+	require.Nil(t, err)
+
+	tests := []struct {
+		description      string
+		key              string
+		optional         bool
+		expectedResolver clortho.Resolver
+		expectedErr      error
+	}{
+		{
+			description:      "Success",
+			key:              "good",
+			optional:         false,
+			expectedResolver: goodResolver,
+		},
+		{
+			description: "Silent failure",
+			key:         "bad",
+			optional:    true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			result := In{}
+			assert := assert.New(t)
+			require := require.New(t)
+			app := fx.New(
+				arrange.TestLogger(t),
+				arrange.ForViper(v),
+				ProvideResolver(tc.key, tc.optional),
+				fx.Invoke(
+					func(in In) {
+						result = in
+					},
+				),
+			)
+			err := app.Err()
+
+			assert.Equal(tc.expectedResolver, result.R)
+			if tc.expectedErr == nil {
+				assert.NoError(err)
+				return
+			}
+			require.Error(err)
+			assert.True(strings.Contains(err.Error(), tc.expectedErr.Error()),
+				fmt.Errorf("error [%v] doesn't contain error [%v]",
+					err, tc.expectedErr),
+			)
+		})
+	}
```

Comment:
I've barely touched fx atm, I'll rely on @johnabass on fx tests while I learn how to use fx!!! 📚